### PR TITLE
Sink and Toilet Fixes+Refactor

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Fixes up sinks and toilets+Refactors them

Most changes from TG

Toilets/Urinals
- Can no longer spam click someone to death using the toilet; standard cooldowns are incurred. Fixes https://github.com/ParadiseSS13/Paradise/issues/3519 
- Damage slightly lowered for toilet head slamming (8->5)
- Sound plays when headslamming someone into a toilet/urinal
- Can fill up open reagent containers with a toilet (ewww).
 - This adds "toilet water" to your reagent container, which functions nearly identically to fishwater.
- Ensures you can't put NO_DROP items in a toilet

Sinks
- No more visible message spam for filling up a container by a sink
- can wash your face or hands in a sink
 - Washing your face makes you less drowsy and washes off lipstick
- Fixes it so you cannot wash abstract items, like grabs
- Can fill up a mop using a sink
- Washing will now use progress bars

:cl: Fox McCloud
bugfix: Can no longer spamclick someone to death using headslamming on a toilet/urimal
tweak: Toilet/urinal headslamming now plays a sound
tweak: Toilet headslamming damage reduced slightly
rscadd: Can now fill open reagent containers in a toilet to receive "toilet water".
tweak: No more message spam when someone fills a container using a sink
rscadd: Can now wash your face using a sink, which washes away lipstick and wakes you up slightly
tweak: Washing things will now use progress bars
/:cl:

### CODE REVIEW

**Comment by Gummin** [comment]
Fox for best plumber 2016! :+1:

**Comment by Fox-McCloud** [comment]
I'm half tempted to make the toilet give you a unique reagent...

**Comment by Gummin** [comment]
I think you should.. :p

**Comment by Fox-McCloud** [comment]
As requested...

Instead of adding water it will now add "toilet water".

I was originally just going to make it fishwater, until I realize that meant you could farm it for botanical purposes, and it wouldn't make much sense from a realism standpoint.

Also fixed an issue with fish water metabolization.

File: code/modules/reagents/oldchem/reagents/reagents_water.dm [hunk 1]
```diff
@@ -223,9 +223,18 @@
...
 	return
 
+/datum/reagent/fishwater/toiletwater
+	name = "Toilet Water"
+	id = "toiletwater"
+	description = "Filthy water scoured from a nasty toilet bowl. Absolutely disgusting."
+	reagent_state = LIQUID
+	color = "#757547"
+
+/datum/reagent/fishwater/toiletwater/reaction_mob(var/mob/M, var/method=TOUCH, var/volume) //For shennanigans
```
Comment (Crazylemon64) [inline code/modules/reagents/oldchem/reagents/reagents_water.dm]: Erm... What sort of shenanigans, exactly?

Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/reagents/reagents_water.dm]: Putting it in someone's drink =p

**Comment by KasparoVy** [comment]
Lmao, excellent. Nice touch with that reagent

File: code/game/objects/structures/watercloset.dm [hunk 2]
```diff
@@ -12,14 +12,18 @@
...
-/obj/structure/toilet/attack_hand(mob/living/user as mob)
+
+/obj/structure/toilet/attack_hand(mob/living/user)
 	if(swirlie)
-		usr.visible_message("<span class='danger'>[user] slams the toilet seat onto [swirlie.name]'s head!</span>", "<span class='notice'>You slam the toilet seat onto [swirlie.name]'s head!</span>", "You hear reverberating porcelain.")
-		swirlie.adjustBruteLoss(8)
+		user.changeNext_move(CLICK_CD_MELEE)
+		playsound(src.loc, "swing_hit", 25, 1)
+		swirlie.visible_message("<span class='danger'>[user] slams the toilet seat onto [swirlie]'s head!</span>", "<span class='userdanger'>[user] slams the toilet seat onto [swirlie]'s head!</span>", "<span class='italics'>You hear reverberating porcelain.</span>")
+		swirlie.adjustBruteLoss(5)
```
Comment (Markolie) [inline code/game/objects/structures/watercloset.dm]: Should probably up this a bit now that there's a cooldown - 10 or 15 maybe.

**Comment by Earthdivine** [comment]
No adding glove recoloring? :sad face:

**Comment by Aurorablade** [comment]
wow, this PR is pretty shitty....*Wait for it...*

**Comment by TheDZD** [comment]
@Aurorablade That's some quality toilet humor.

**Comment by Dumbdumn5** [comment]
My God, we've done it. We've gone as far as to balance literal shit containers. WE'VE GONE TOO FAR! (One less spam clickable bug!)

**Comment by KasparoVy** [comment]
>drunk as duck, about to pass out
>WASHES FACE INTENSELY
>wakes right up, good as new

**Pull request merged**